java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -0,0 +1,1663 @@
+      }
+    }
+    return result;
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_getNativeOffsetsLength(JNIEnv *env, jclass,
+                                                                              jlong handle) {
+  JNI_NULL_CHECK(env, handle, "native handle is null", 0);
+  try {
+    cudf::jni::auto_set_device(env);
+    jlong result = 0;
+    cudf::column_view *column = reinterpret_cast<cudf::column_view *>(handle);
+    if (column->type().id() == cudf::type_id::STRING) {
+      if (column->size() > 0) {
+        cudf::strings_column_view view = cudf::strings_column_view(*column);
+        cudf::column_view offsets_view = view.offsets();
+        result = sizeof(int) * offsets_view.size();
+      }
+    } else if (column->type().id() == cudf::type_id::LIST) {
+      if (column->size() > 0) {
+        cudf::lists_column_view view = cudf::lists_column_view(*column);
+        cudf::column_view offsets_view = view.offsets();
+        result = sizeof(int) * offsets_view.size();
+      }
+    }
+    return result;
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_getNativeValidityAddress(JNIEnv *env,
+                                                                                jclass,
+                                                                                jlong handle) {
+  JNI_NULL_CHECK(env, handle, "native handle is null", 0);
+  try {
+    cudf::jni::auto_set_device(env);
+    cudf::column_view *column = reinterpret_cast<cudf::column_view *>(handle);
+    return reinterpret_cast<jlong>(column->null_mask());
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_getNativeValidityLength(JNIEnv *env,
+                                                                               jclass,
+                                                                               jlong handle) {
+  JNI_NULL_CHECK(env, handle, "native handle is null", 0);
+  try {
+    cudf::jni::auto_set_device(env);
+    cudf::column_view *column = reinterpret_cast<cudf::column_view *>(handle);
+    jlong result = 0;
+    if (column->null_mask() != nullptr) {
+      result = cudf::bitmask_allocation_size_bytes(column->size());
+    }
+    return result;
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_getNativeValidPointerSize(JNIEnv *env,
+                                                                                 jobject j_object,
+                                                                                 jint size) {
+  try {
+    cudf::jni::auto_set_device(env);
+    return static_cast<jlong>(cudf::bitmask_allocation_size_bytes(size));
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_getDeviceMemorySize(JNIEnv *env, jclass,
+                                                                             jlong handle) {
```

Comment:
Nit: indentation is off here.

Reply:
I just cant seem to get Intellij to do the right thing with jni files. Thanks for the catch :) Fixed.